schematic_api/api/routes.py

```diff
@@ -620,12 +672,14 @@ def download_manifest(input_token, dataset_id, asset_view, as_json, new_manifest
     manifest_data = store.getDatasetManifest(datasetId=dataset_id, downloadFile=True, newManifestName=new_manifest_name)
 
     #return local file path
-    manifest_local_file_path = manifest_data['path']
+    try:
```

Comment:
Can you explain what is going on here?  So, I'm assuming by reading the code that if the key `path` doesn't exist, the manifest wasn't downloaded. 

It's best practice to throw the `KeyError` then:

`except KeyError:`

Reply:
Yes. Raising a key error here makes sense. Alternatively, I could use the .get function: 
```
manifest_data.get("path", "")
``` 
But since not getting the path key from manifest_data would be problematic, so maybe raising an error is better?

Reply:
Try something like this: https://stackoverflow.com/questions/9157210/how-do-i-raise-the-same-exception-with-a-custom-message-in-python.  Right now, i think your code will fail.

```
except KeyError as e:
     raise KeyError(f'Failed to download manifest from dataset: {dataset_id}') from e
```
OR

```
except KeyError as e:
   logging.error(f'Failed to download manifest from dataset: {dataset_id}')
   raise e
```

In the future, you can imagine having custom exceptions for `Schematic` that make error catching and raising more robust.

Reply:
Thanks Tom. We do have exception.py (see the document [here](https://github.com/Sage-Bionetworks/schematic/blob/develop-light-refactor-download-manifest/schematic/exceptions.py)). Do you think if i should create another class there for handling login error? I also noticed that we have `AccessCredentialsError` class (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop-light-refactor-download-manifest/schematic/exceptions.py#:~:text=class%20AccessCredentialsError(Exception))), but that one seems to only for handling not having access to a certain entity in synapse, and not really login error.

Reply:
I wouldn't worry about it in this particular PR, but just make note of it as a possible update in the future. What you're doing a lot here is catching and re-raising the exception for "better error messages". I don't think you need to try to catch every single error, sometimes its ok for functions to just fail, but in this particular `KeyError` scenario, I was referring to now this particular syntax:

```
raise()
```

Will fail without an Exception type.

Reply:
Thanks Tom. I have updated the syntax there. I think instead of raising a KeyError, it makes more sense to raise a type error because manifest_data can be an empty string when it returns (and `manifest_data["path"]` would therefore not work)